xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/serverinfo/ClioServerInfo.java

```diff
@@ -0,0 +1,186 @@
+
+/**
+ * An implementation of {@link ServerInfo} that conforms to Clio server payloads.
+ */
+@Value.Immutable
+@JsonSerialize(as = ImmutableClioServerInfo.class)
+@JsonDeserialize(as = ImmutableClioServerInfo.class)
+public interface ClioServerInfo extends ServerInfo {
+
+  /**
+   * Construct a builder for this class.
+   *
+   * @return An {@link ImmutableClioServerInfo.Builder}.
+   */
+  static ImmutableClioServerInfo.Builder builder() {
+    return ImmutableClioServerInfo.builder();
+  }
+
+  /**
+   * The version number of the running clio version.
+   *
+   * @return A {@link String} containing the version number.
+   */
+  @JsonProperty("clio_version")
+  String clioVersion();
+
+  /**
+   * The version number of the running rippled version.
+   *
+   * @return A {@link String} containing the version number.
+   */
+  @JsonProperty("rippled_version")
+  String rippledVersion();
+
+  /**
+   * Range expression indicating the sequence numbers of the ledger versions the local rippled has in its database. This
+   * may be a disjoint sequence such as {@code 24900901-24900984,24901116-24901158}. If the server does not have any
+   * complete ledgers (for example, it recently started syncing with the network), this will be an empty
+   * {@link String}.
+   *
+   * @return A {@link String} representing a range of ledger sequences.
+   */
+  @JsonProperty("complete_ledgers")
+  String completeLedgers();
+
+  /**
+   * Accessor for a range of ledgers.
+   *
+   * @return A {@link List} of type {@link Range} of type {@link UnsignedLong}.
+   */
+  @Value.Derived
+  @JsonIgnore
+  default List<Range<UnsignedLong>> completeLedgerRanges() {
+    return LedgerRangeUtils.completeLedgerRanges(completeLedgers());
+  }
+
+  /**
+   * Determines if the supplied {@code ledgerIndex} exists on the rippled server by inspecting
+   * {@link #completeLedgers()}.
+   *
+   * @param ledgerIndex An {@link UnsignedLong} representing a particular ledger index.
+   *
+   * @return {@code true} if the rippled server includes {@code ledgerIndex} in its local database; {@code false}
+   *   otherwise.
+   */
+  @Value.Derived
+  @JsonIgnore
+  default boolean isLedgerInCompleteLedgers(final UnsignedLong ledgerIndex) {
+    return this.completeLedgerRanges().stream()
+      .anyMatch(range -> range.contains(ledgerIndex));
+  }
+
+  /**
+   * The load-scaled open ledger transaction cost the server is currently enforcing, as a multiplier on the base
+   * transaction cost. For example, at 1000 load factor and a reference transaction cost of 10 drops of XRP, the
+   * load-scaled transaction cost is 10,000 drops (0.01 XRP). The load factor is determined by the highest of the
+   * individual server's load factor, the cluster's load factor, the open ledger cost and the overall network's load
+   * factor.
+   *
+   * @return An optionally-present {@link BigDecimal} representing the load factor.
+   */
+  @JsonProperty("load_factor")
+  Optional<BigDecimal> loadFactor();
```

Comment:
How do we know these fields should be `Optional` vs the fields above that are not?